### PR TITLE
move the add/edit description button to the app-bar

### PR DESCRIPTION
**Pull request by mawinter69**
So far the add/edit description button is rendered on its own line to the far right. This leads to additional space between the description or the title and the following content.

This change moves the add/edit description button to the app-bar where applicable (jobs, runs, computers, users and labels). For views it stays as it is as they don't have a title/app-bar. One also can't just allow to change the description via the configure page as the `All` View has no configure page.

The change is backwards compatible so plugins using the editableDescription taglib will work as before.

Requires changes in Matrix-project and maven-project plugins to align the new layout

#### Job:
Before:
image
After:
image
A pipeline job:
image

#### Run:
Before:
image
After:
image
After clicking the `Add Description` button
image

#### User:
Before:
image
After:
image

#### Computer:
Before:
image
After:
image

#### Label:
Before:
image
After:
image

### Testing done
Manual Testing

### Proposed changelog entries

- Move Add/Edit description to app bar.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Comment by mawinter69** [comment]
Some questions to think about
- should I directly create an overflow button that takes add/edit description

On the run page
- also move the keep log forever to overflow
- move the info about when the build started and how long it took to the left like e.g. the info about the cause.

image

**Comment by mawinter69** [comment]
@jenkinsci/sig-ux

**Comment by janfaracik** [comment]
Like it 👍

> * make the add/edit description button a secondary button?

Seems sensible yeah.

> * should I directly create an overflow button that takes add/edit description?

Could do, I do wonder how much benefit/use the 'Add description' button gets to warrant it existing - rather than just existing in the 'Configure' page.

> * also move the keep build forever to overflow?

Sounds good.

> * move the info about when the build started and how long it took to the left like e.g. the info about the cause.

Want to eventually move this section into its own extendable card (in a similar fashion to https://jenkins-redesign.vercel.app/project/build/375). I don't think the current summary components are the best use of space.

File: core/src/main/resources/hudson/model/Job/index.jelly
Comment (yaroslavafenkin) [inline core/src/main/resources/hudson/model/Job/index.jelly:44]: There's no `Job.UPDATE`, is there? Is there any reason you went for a change?

File: war/src/main/webapp/scripts/hudson-behavior.js
Comment (yaroslavafenkin) [inline war/src/main/webapp/scripts/hudson-behavior.js:1904]:
```
hudson-behavior.js:1904 Uncaught TypeError: Cannot read properties of null (reading 'firstElementChild')
    at replaceDescription (hudson-behavior.js:1904:18)
    at HTMLAnchorElement.<anonymous> (editable-description.js:10:14)
```
when I hit edit description at `/manage/computer/(built-in)/`

**Comment by mawinter69** [comment]
> Like it 👍
> 
> > * make the add/edit description button a secondary button?
> 
> Seems sensible yeah.
> 
> > * should I directly create an overflow button that takes add/edit description?
> 
> Could do, I do wonder how much benefit/use the 'Add description' button gets to warrant it existing - rather than just existing in the 'Configure' page.
> 
> > * also move the keep build forever to overflow?
> 
> Sounds good.
> 
> > * move the info about when the build started and how long it took to the left like e.g. the info about the cause.
> 
> Want to eventually move this section into its own extendable card (in a similar fashion to https://jenkins-redesign.vercel.app/project/build/375). I don't think the current summary components are the best use of space.

For now I only made the description button a secondary button.

File: core/src/main/resources/hudson/model/Job/index.jelly
Comment (mawinter69) [reply core/src/main/resources/hudson/model/Job/index.jelly]: thanks fixed

File: war/src/main/webapp/scripts/hudson-behavior.js
Comment (mawinter69) [reply war/src/main/webapp/scripts/hudson-behavior.js:1904]: ahh the built-in has no description. Worked around by only showing the button when there is an actual div with id `description`

**Comment by Kevin-CB** [comment]
LGTM security wise

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
